### PR TITLE
Resolves #20: Additional map decoration

### LINKED ISSUE #20: Additional map decoration
The first "layer" of decoration around the map view is already there but the second one, showing the Coordinates and highlighting the MainCoordinates is still missing. [The mockup](https://raw.githubusercontent.com/Baret/pltcmd/master/mockups/mapview.png) shows how it might look like. But maybe it is better to center the coordinate values.

### PR DESCRIPTION
**Pull request by Loomie**

### CODE REVIEW

File: src/main/kotlin/de/gleex/pltcmd/ui/renderers/MapCoordinatesDecorationRenderer.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.data.Size
+import org.hexworks.zircon.api.graphics.impl.SubTileGraphics
+import de.gleex.pltcmd.model.world.WorldMap
+import de.gleex.pltcmd.game.GameWorld
+
+/** Draws major grid coordinates around a component. It will be displayed on the top and left side. Designed to work with [MapGridDecorationRenderer]. */
+class MapCoordinateDecorationRenderer(private val world: GameWorld) : ComponentDecorationRenderer {
+
+    // top, left, bottom, right
```
Comment (Baret) [inline src/main/kotlin/de/gleex/pltcmd/ui/renderers/MapCoordinatesDecorationRenderer.kt]: What does this comment mean?

File: src/main/kotlin/de/gleex/pltcmd/game/GameWorld.kt [hunk 2]
```diff
@@ -39,4 +39,11 @@ class GameWorld(worldMap: WorldMap): GameArea<Tile, MapBlock> by GameAreaBuilder
     }
 
     private fun Coordinate.toPosition() = Position3D.create(eastingFromLeft, northingFromBottom, 0)
+
+    private fun Position3D.toCoordinate() = Coordinate(x, y)
```
[the remark this answers is not in the document]
Comment (Loomie) [reply src/main/kotlin/de/gleex/pltcmd/game/GameWorld.kt]: Yes.

File: src/main/kotlin/de/gleex/pltcmd/ui/renderers/MapCoordinatesDecorationRenderer.kt (hunk 1 above)
Comment (Loomie) [reply src/main/kotlin/de/gleex/pltcmd/ui/renderers/MapCoordinatesDecorationRenderer.kt]: It's a copy of `MapGridDecorationRenderer` and wrong here

File: src/main/kotlin/de/gleex/pltcmd/ui/renderers/MapCoordinates.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.hexworks.zircon.api.data.Tile
+import org.hexworks.zircon.api.graphics.StyleSet
+import org.hexworks.zircon.api.graphics.Symbols
+import org.hexworks.zircon.api.graphics.TileGraphics
+import org.hexworks.zircon.api.resource.TilesetResource
+import org.hexworks.zircon.api.shape.LineFactory
+import de.gleex.pltcmd.game.GameWorld
+import de.gleex.pltcmd.model.world.Coordinate
+import org.hexworks.zircon.api.CharacterTileStrings
+import org.hexworks.zircon.api.builder.data.TileBuilder
```
[the remark this answers is not in the document]
Comment (Loomie) [reply src/main/kotlin/de/gleex/pltcmd/ui/renderers/MapCoordinates.kt]: Doesn't work here with Eclipse :(

**Pull request merged**